### ISSUE TITLE
Apply label to issues with open dependencies

### ISSUE THREAD

**Issue by jcommelin** [jonabc/sync-task-issues#20]
Would it be possible to automatically add a label to issues/PRs with open dependencies? (And auto-remove that label when all dependencies are closed?)

**Comment by jonabc** [comment]
@jcommelin thats an interesting idea, but I don't think that would be something to handle in this action directly.

IIUC, you're wanting to detect when all of a "parent" issue's linked "child" issues/PRs are closed. In that case the unit of action is the parent.  This action deals with the inverse, detecting a change to an individual "child" issue or PR and updating each parent reference.  In order to keep this action simple and composable for a workflow file, it's out of scope to do further calculations or actions on the parents.

I can add as output to this action every detected parent reference that was operated on and the action that was taken, i.e. marked complete or incomplete.  A followup step could then check those issues and see if all of the references are closed/open, what labels are present, whatever else is desired, and add or remove labels accordingly.  That followup step could be implemented using something like [action/github-script](https://github.com/actions/github-script) for one-offs, or written as it's own action if it needed to be reused and made publicly available for others.

I'll leave this open to track adding some outputs to the action that might be helpful.

**Comment by jcommelin** [comment]
Thanks for the fast reply. Such output would already be very helpful. I don't have any experience with writing actions myself, but I can probably hack something together.

The use case: easily filter for PRs that are waiting for a review, but are not blocked by other open PRs.

**Comment by jonabc** [comment]
Ah cool, 👍  I'll try to get that updated today or tomorrow.  Or, if you wanted to open a PR for the change I'd be happy to review it 🙏

**Comment by jcommelin** [comment]
Thanks for your very fast action!
